src/Maker/MakeForgottenPassword.php

```diff
@@ -0,0 +1,311 @@
+                'password_setter' => $input->getArgument('password-setter'),
+                'login_route' => 'app_login',
+                'token_class_name' => $tokenClassNameDetails->getShortName(),
+                'token_full_class_name' => $tokenClassNameDetails->getFullName(),
+            ]
+        );
+
+        // 5) Generate the "request" template
+        $generator->generateFile(
+            'templates/forgottenPassword/request.html.twig',
+            'forgottenPassword/twig_request.tpl.php',
+            [
+                'email_field' => $emailField,
+            ]
+        );
+
+        // 6) Generate the reset e-mail template
+        $generator->generateFile(
+            'templates/forgottenPassword/email.txt.twig',
+            'forgottenPassword/twig_email.tpl.php',
+            []
+        );
+
+        // 7) Generate the "checkEmail" template
+        $generator->generateFile(
+            'templates/forgottenPassword/check_email.html.twig',
+            'forgottenPassword/twig_check_email.tpl.php',
+            []
+        );
+
+        // 8) Generate the "reset" template
+        $generator->generateFile(
+            'templates/forgottenPassword/reset.html.twig',
+            'forgottenPassword/twig_reset.tpl.php',
+            []
+        );
+
+        $generator->writeChanges();
+        $this->writeSuccessMessage($io);
+
+        $io->text('Done! A new entity was added: PasswordResetToken. You should now generate a migration (make:migration) and run it to update your database.');
+        $io->text('Next: Please review ForgottenPasswordController. Then you can add a link to "app_forgotten_password" path anywhere you like, typically below your login form!');
+    }
+
+    private function generateRequestFormClass(Generator $generator, string $emailField)
+    {
+        $formClassDetails = $generator->createClassNameDetails(
+            'PasswordRequestFormType',
+            'Form\\'
+        );
+
+        $formFields = [
+            $emailField => [
+                'type' => TextType::class,
+                'options_code' => <<<EOF
+                'constraints' => [
+                    new NotBlank([
+                        'message' => 'Please enter your $emailField',
+                    ]),
+                ],
+EOF
+            ],
+        ];
+
+        $this->formTypeRenderer->render(
+            $formClassDetails,
+            $formFields,
+            null,
+            [
+                'Symfony\Component\Validator\Constraints\NotBlank',
+            ]
+        );
+
+        return $formClassDetails;
+    }
+
+    private function generateResettingFormClass(ClassNameDetails $userClassDetails, Generator $generator)
+    {
+        $formClassDetails = $generator->createClassNameDetails(
+            'PasswordResettingFormType',
+            'Form\\'
+        );
+
+        $formFields = [
+            'plainPassword' => [
+                'type' => PasswordType::class,
```

Comment:
Don't we want to use `RepeatedType` here?

Reply:
`RepeatedType` is not used for Maker registration form, so I kept doing the same.  But I agree we should use it at both places.

Reply:
Actually, I don't think we need it for the registration form (it gives you rapid registration), but we do need it for password reset I think. And I've noticed this strategy on most websites

Reply:
Ok, I will modify!